libs-scala/grpc-reverse-proxy/src/main/scala/com/daml/grpc/ForwardService.scala

```diff
@@ -0,0 +1,26 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.grpc
+
+import java.io.{ByteArrayInputStream, InputStream}
+
+import com.daml.grpc.reflection.ServiceDescriptorInfo
+import com.google.common.io.ByteStreams
+import io.grpc.{CallOptions, Channel, MethodDescriptor, ServerServiceDefinition}
+
+private[grpc] object ForwardService {
+
+  def apply(backend: Channel, service: ServiceDescriptorInfo): ServerServiceDefinition =
+    service.methods
+      .map(_.toMethodDescriptor(ByteArrayMarshaller, ByteArrayMarshaller))
+      .map(ForwardCall(_, backend, CallOptions.DEFAULT))
+      .foldLeft(ServerServiceDefinition.builder(service.fullServiceName))(_ addMethod _)
+      .build()
+
+  private object ByteArrayMarshaller extends MethodDescriptor.Marshaller[Array[Byte]] {
+    override def parse(input: InputStream): Array[Byte] = ByteStreams.toByteArray(input)
```

Comment:
I realize you might have copied that from my hacked up prototype and I know that I did it this way because I couldn’t figure out another way but let me ask anyway: Can we stream this instead of materializing the input and output fully? It seems like it should work but I couldn’t quite figure out how to plug it together.

Reply:
I asked myself the first question, but without copying I had weird errors when running the conformance test suite on the proxied sandbox.

Reply:
yeah that sounds similar to what I was seeing. it looked like something might have been too lazy and was leaking resources but I couldn’t quite pin it down or fix it. Totally fine to ignore it for now.

On a related note since you mentioned it: should we have a test that runs the ledger api test tool against the proxied sandbox? Doesn’t have to be in this PR.

Reply:
I considered it. Does it make sense to add it here? Or does it make more sense (although it's more complicated) to have this as an integration test on the non-repudiation middleware?

Reply:
Just adding it to the non-repudiation middleware seems sufficient. I guess if we really want to we could also add it to both but I don’t see too much benefit.

Reply:
If the reverse proxy grows in scope it will probably make sense to add more tests. for sure.